src/chai/managed_ptr.hpp

```diff
@@ -272,7 +272,7 @@ namespace chai {
             m_gpu_pointer(other.m_gpu_pointer),
             m_pointer_record(other.m_pointer_record)
          {
-#ifndef __CUDA_ARCH__
+#if !defined(__CUDA_ARCH__) && !defined(__HIP_DEVICE_COMPILE__)
```

Comment:
Can you come up with another more generic macro for this case?

Reply:
Done. I added another macro. 

I tested it by inserting the following code block into the code:

#if !defined(__CUDA_ARCH__) && !defined(__HIP_DEVICE_COMPILE__)
#warning compiling host previous macro
#else
#warning compiling device previous macro
#endif

#if !defined(__DEVICE_COMPILE__)
#warning compiling host danny macro
#else
#warning compiling device  danny macro
#endif

And it came out consistent:

/usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/tpl/chai/src/chai/managed_ptr.hpp:282:2: warning: compiling device previous macro [-W#warnings]
#warning compiling device previous macro
 ^
/usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/tpl/chai/src/chai/managed_ptr.hpp:288:2: warning: compiling device  danny macro [-W#warnings]
#warning compiling device  danny macro
 ^
2 warnings generated when compiling for gfx900.
In file included from /usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/src/care/RAJAPlugin.cpp:12:
In file included from /usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/src/care/util.h:12:
In file included from /usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/src/care/care.h:267:
In file included from /usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/src/care/PointerTypes.h:30:
/usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/tpl/chai/src/chai/managed_ptr.hpp:280:2: warning: compiling host previous macro [-W#warnings]
#warning compiling host previous macro
 ^
/usr/workspace/taller1/CARE_TESTING_CORONA_2/CARE/tpl/chai/src/chai/managed_ptr.hpp:286:2: warning: compiling host danny macro [-W#warnings]
#warning compiling host danny macro